backend/services/monthly_dump_service.py

```
@@ -0,0 +1,350 @@
+            # SSRF Protection: Resolve and check IP
+            ip_info = socket.getaddrinfo(hostname, None)
+            for item in ip_info:
+                ip_str = item[4][0]
+                # Filter out IPv6 wrappers or malformed addresses
+                if "%" in ip_str:
+                    ip_str = ip_str.split("%")[0]
+                ip = ipaddress.ip_address(ip_str)
+                if ip.is_private or ip.is_loopback:
+                    return False
+            return True
+        except Exception:
```

Ruff (0.15.10)

[warning] Do not catch blind exception: `Exception`

(BLE001)